File: src/main/java/org/apache/fineract/client/services/SurveyApi.java
/**
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements. See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership. The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License. You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied. See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */
package org.apache.fineract.client.services;

import org.apache.fineract.client.CollectionFormats.*;

import io.reactivex.Observable;
import io.reactivex.Completable;
import retrofit2.Call;
import retrofit2.http.*;

import okhttp3.RequestBody;
import okhttp3.ResponseBody;
import okhttp3.MultipartBody;

import org.apache.fineract.client.models.GetSurveyResponse;
import org.apache.fineract.client.models.PostSurveySurveyNameApptableIdRequest;
import org.apache.fineract.client.models.PostSurveySurveyNameApptableIdResponse;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

public interface SurveyApi {
  /**
   * Create an entry in the survey table
   * Insert and entry in a survey table (full fill the survey).  Refer Link for sample Body:  [ https://demo.fineract.dev/fineract-provider/api-docs/apiLive.htm#survey_create ] 
   * @param surveyName surveyName (required)
   * @param apptableId apptableId (required)
   * @param postSurveySurveyNameApptableIdRequest  (required)
   * @return Call&lt;PostSurveySurveyNameApptableIdResponse&gt;
   */
  @Headers({
    "Content-Type:application/json"
  })
  @POST("survey/{surveyName}/{apptableId}")
  Call<PostSurveySurveyNameApptableIdResponse> createDatatableEntry1(
    @retrofit2.http.Path("surveyName") String surveyName, @retrofit2.http.Path("apptableId") Long apptableId, @retrofit2.http.Body PostSurveySurveyNameApptableIdRequest postSurveySurveyNameApptableIdRequest
  );

  /**
   * 
   * 
   * @param surveyName  (required)
   * @param clientId  (required)
   * @param fulfilledId  (required)
   * @return Call&lt;String&gt;
   */
  @DELETE("survey/{surveyName}/{clientId}/{fulfilledId}")
  Call<String> deleteDatatableEntries2(
    @retrofit2.http.Path("surveyName") String surveyName, @retrofit2.http.Path("clientId") Long clientId, @retrofit2.http.Path("fulfilledId") Long fulfilledId
  );

  /**
   * 
   * 
   * @param surveyName  (required)
   * @param clientId  (required)
   * @return Call&lt;String&gt;
   */
  @GET("survey/{surveyName}/{clientId}")
  Call<String> getClientSurveyOverview(
    @retrofit2.http.Path("surveyName") String surveyName, @retrofit2.http.Path("clientId") Long clientId
  );

  /**
   * 
   * 
   * @param surveyName  (required)
   * @param clientId  (required)
   * @param entryId  (required)
   * @return Call&lt;String&gt;
   */
  @GET("survey/{surveyName}/{clientId}/{entryId}")
  Call<String> getSurveyEntry(
    @retrofit2.http.Path("surveyName") String surveyName, @retrofit2.http.Path("clientId") Long clientId, @retrofit2.http.Path("entryId") Long entryId
  );

  /**
   * 
   * 
   * @param surveyName  (required)
   * @param apptable  (required)
   * @param body  (optional)
   * @return Call&lt;String&gt;
   */
  @Headers({
    "Content-Type:application/json"
  })
  @PUT("survey/register/{surveyName}/{apptable}")
  Call<String> register(
    @retrofit2.http.Path("surveyName") String surveyName, @retrofit2.http.Path("apptable") String apptable, @retrofit2.http.Body String body
  );

  /**
   * Retrieve survey
   * Lists a registered survey table details and the Apache Fineract Core application table they are registered to.
   * @param surveyName surveyName (required)
   * @return Call&lt;GetSurveyResponse&gt;
   */
  @GET("survey/{surveyName}")
  Call<GetSurveyResponse> retrieveSurvey(
    @retrofit2.http.Path("surveyName") String surveyName
  );

  /**
   * Retrieve surveys
   * Retrieve surveys. This allows to retrieve the list of survey tables registered .
   * @return Call&lt;List&lt;GetSurveyResponse&gt;&gt;
   */
  @GET("survey")
  Call<List<GetSurveyResponse>> retrieveSurveys();
    

}

File: src/main/kotlin/org/openapitools/client/apis/SurveyApi.kt
package org.openapitools.client.apis

import org.openapitools.client.models.GetSurveyResponse
import org.openapitools.client.models.PostSurveySurveyNameApptableIdRequest
import org.openapitools.client.models.PostSurveySurveyNameApptableIdResponse
import retrofit2.http.Body
import retrofit2.http.DELETE
import retrofit2.http.GET
import retrofit2.http.POST
import retrofit2.http.PUT
import retrofit2.http.Path

interface SurveyApi {
    /**
     * Create an entry in the survey table
     * Insert and entry in a survey table (full fill the survey).  Refer Link for sample Body:  [ https://fineract.apache.org/legacy-docs/apiLive.htm#survey_create ]
     * Responses:
     *  - 200: OK
     *
     * @param surveyName surveyName
     * @param apptableId apptableId
     * @param postSurveySurveyNameApptableIdRequest
     * @return [PostSurveySurveyNameApptableIdResponse]
     */
    @POST("v1/survey/{surveyName}/{apptableId}")
    suspend fun createDatatableEntry1(
        @Path("surveyName") surveyName: String,
        @Path("apptableId") apptableId: Long,
        @Body postSurveySurveyNameApptableIdRequest: PostSurveySurveyNameApptableIdRequest
    ): PostSurveySurveyNameApptableIdResponse

    /**
     *
     *
     * Responses:
     *  - 0: default response
     *
     * @param surveyName
     * @param clientId
     * @param fulfilledId
     * @return [kotlin.String]
     */
    @DELETE("v1/survey/{surveyName}/{clientId}/{fulfilledId}")
    suspend fun deleteDatatableEntries1(
        @Path("surveyName") surveyName: String,
        @Path("clientId") clientId: Long,
        @Path("fulfilledId") fulfilledId: Long
    ): String

    /**
     *
     *
     * Responses:
     *  - 0: default response
     *
     * @param surveyName
     * @param clientId
     * @return [kotlin.String]
     */
    @GET("v1/survey/{surveyName}/{clientId}")
    suspend fun getClientSurveyOverview(
        @Path("surveyName") surveyName: String,
        @Path("clientId") clientId: Long
    ): String

    /**
     *
     *
     * Responses:
     *  - 0: default response
     *
     * @param surveyName
     * @param clientId
     * @param entryId
     * @return [kotlin.String]
     */
    @GET("v1/survey/{surveyName}/{clientId}/{entryId}")
    suspend fun getSurveyEntry(
        @Path("surveyName") surveyName: String,
        @Path("clientId") clientId: Long,
        @Path("entryId") entryId: Long
    ): String

    /**
     *
     *
     * Responses:
     *  - 0: default response
     *
     * @param surveyName
     * @param apptable
     * @param body  (optional)
     * @return [kotlin.String]
     */
    @PUT("v1/survey/register/{surveyName}/{apptable}")
    suspend fun register(
        @Path("surveyName") surveyName: String,
        @Path("apptable") apptable: String,
        @Body body: String? = null
    ): String

    /**
     * Retrieve survey
     * Lists a registered survey table details and the Apache Fineract Core application table they are registered to.
     * Responses:
     *  - 200: OK
     *
     * @param surveyName surveyName
     * @return [GetSurveyResponse]
     */
    @GET("v1/survey/{surveyName}")
    suspend fun retrieveSurvey(@Path("surveyName") surveyName: String): GetSurveyResponse

    /**
     * Retrieve surveys
     * Retrieve surveys. This allows to retrieve the list of survey tables registered .
     * Responses:
     *  - 200: OK
     *
     * @return [kotlin.collections.List<GetSurveyResponse]
     */
    @GET("v1/survey")
    suspend fun retrieveSurveys(): List<GetSurveyResponse>

}
